utils/zerovec/src/yoke_impls.rs

```diff
@@ -104,3 +96,40 @@ where
         unsafe { f(mem::transmute::<&mut Self, &mut Self::Output>(self)) }
     }
 }
+
+impl<'a, T: 'static + AsULE + ?Sized> ZeroCopyFrom<ZeroVec<'a, T>> for ZeroVec<'static, T> {
+    fn zero_copy_from<'b>(cart: &'b ZeroVec<'a, T>) -> ZeroVec<'b, T> {
+        ZeroVec::Borrowed(cart.as_slice())
+    }
+}
+
+impl<'a, T: 'static + AsVarULE + Clone> ZeroCopyFrom<VarZeroVec<'a, T>> for VarZeroVec<'static, T> {
+    fn zero_copy_from<'b>(cart: &'b VarZeroVec<'a, T>) -> VarZeroVec<'b, T> {
+        // the owned variant is not compatible with the borrowed one
+        // clones are shallow for the borrowed variant anyway
+        Clone::clone(cart)
+    }
+}
+
+impl<'a, K, V> ZeroCopyFrom<ZeroMap<'a, K, V>> for ZeroMap<'static, K, V>
+where
+    K: 'static + for<'b> ZeroMapKV<'b>,
+    V: 'static + for<'b> ZeroMapKV<'b>,
+    <K as ZeroMapKV<'static>>::Container: for<'b> ZeroCopyFrom<<K as ZeroMapKV<'b>>::Container>,
+    <V as ZeroMapKV<'static>>::Container: for<'b> ZeroCopyFrom<<V as ZeroMapKV<'b>>::Container>,
+    <K as ZeroMapKV<'static>>::Container:
+        for<'b> Yokeable<'b, Output = <K as ZeroMapKV<'b>>::Container>,
+    <V as ZeroMapKV<'static>>::Container:
+        for<'b> Yokeable<'b, Output = <V as ZeroMapKV<'b>>::Container>,
```

Comment:
Question: Did you check that this impl compiles at the call site and doesn't hit https://github.com/rust-lang/rust/issues/85636?

Reply:
It does, at least in the ways I'm trying it. That bug is pretty subtle; if you tweak things a little bit it stops occurring (and more tweaks make it recur).

But also I don't think this _matters_ until we try to use this; I'm just fixing up the impl early.